addons/account_peppol/wizard/peppol_registration.py

```diff
@@ -91,25 +127,69 @@ def _onchange_phone_number(self):
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
+    @api.depends('company_id')
+    def _compute_parent_company_id(self):
+        for wizard in self:
+            wizard.parent_company_id = wizard.company_id.peppol_parent_company_id or wizard.company_id
+
+    @api.depends('company_id')
+    def _compute_display_choice_of_company(self):
+        for wizard in self:
+            wizard.display_choice_of_company = wizard.company_id != wizard.parent_company_id
+
+    @api.depends('company_id')
+    def _compute_choice_of_company(self):
+        for wizard in self:
+            if wizard.company_id == wizard.parent_company_id:
+                wizard.choice_of_company = 'use_self'
+            else:
+                wizard.choice_of_company = 'use_parent' if wizard.parent_company_id.peppol_can_send else 'use_self'
+
+    @api.depends('choice_of_company', 'peppol_eas', 'peppol_endpoint')
+    def _compute_registration_mode(self):
+        for wizard in self:
+            if (
+                wizard.company_id != wizard.parent_company_id
+                and wizard.peppol_eas == wizard.parent_company_id.peppol_eas
+                and wizard.peppol_endpoint == wizard.parent_company_id.peppol_endpoint
+            ):
+                wizard.registration_mode = 'register_as_parent_manually'
+            else:
+                wizard.registration_mode = 'register_as_self'
+
+    @api.depends('registration_mode')
+    @api.depends_context('company')
+    def _compute_selected_company_id(self):
+        for wizard in self:
+            if wizard.registration_mode == 'register_as_parent':
+                wizard.selected_company_id = wizard.parent_company_id
+            else:
+                wizard.selected_company_id = wizard.company_id
+
     @api.depends('company_id.account_edi_proxy_client_ids')
     def _compute_edi_user_id(self):
         for wizard in self:
             wizard.edi_user_id = wizard.company_id.account_edi_proxy_client_ids.filtered(lambda u: u.proxy_type == 'peppol')[:1]
 
-    @api.depends('peppol_eas', 'peppol_endpoint', 'smp_registration')
+    @api.depends('peppol_eas', 'peppol_endpoint', 'registration_mode', 'smp_registration')
     def _compute_peppol_warnings(self):
         for wizard in self:
             peppol_warnings = {}
             if (
                 wizard.peppol_eas
                 and wizard.peppol_endpoint
-                and not wizard.company_id._check_peppol_endpoint_number(warning=True)
+                and not wizard.selected_company_id._check_peppol_endpoint_number(warning=True)
```

Comment:
it is not in the dependencies (even if I don't see when it could cause an issue)